src/components/ContactModal.tsx

```diff
@@ -110,29 +88,39 @@ export const UpsertContactModal: FC<{
               })}
               placeholder="Enter contact's name"
             />
-            {errors.name && <FormErrorMessage>{errors.name.message}</FormErrorMessage>}
+            {errors.name && (
+              <FormErrorMessage data-testid="name-error">{errors.name.message}</FormErrorMessage>
+            )}
           </FormControl>
           <FormControl isInvalid={!!errors.pkh} marginY={5}>
             <FormLabel>Address</FormLabel>
             <Input
               type="text"
               {...register("pkh", {
                 required: "Address is required",
-                validate: validatePkh,
+                validate: isEdit ? () => true : validatePkh,
```

Comment:
well, it's an optimisation, but not very needed tbh. we won't gain much performance gain. it's fine to always validate it

Reply:
Without this fix, the check fails for edit case - cause pkh already exists in contacts 🙂